### PR TITLE
Added tests for extra event functions

### PR DESCRIPTION
**Pull request by MICHAELMUNAVU83**
In this PR , I added tests for the following functions `get_all_available_tickets` , `get_elixir_conf_event `  , `get_elixir_conf_event_and_ticket_types`

### CODE REVIEW

File: lib/elixir_conf_africa/events.ex [hunk 1]
```diff
@@ -27,7 +27,8 @@ defmodule ElixirConfAfrica.Events do
     |> Repo.preload(:ticket_types)
   end
 
-  defp get_elixir_conf_event do
+  @spec get_elixir_conf_event() :: any()
```
Comment (okothkongo) [inline lib/elixir_conf_africa/events.ex:30]:
```suggestion
  @spec get_elixir_conf_event() :: nil|Event.t()
```

**Review by okothkongo — CHANGES_REQUESTED** [review]

File: test/elixir_conf_africa/events_test.exs [hunk 2]
```diff
@@ -22,11 +24,75 @@ defmodule ElixirConfAfrica.EventsTest do
...
+      valid_attrs = %{
+        name: "ElixirConf Africa 2024",
+        description: "some description",
+        location: "some location",
+        event_type: "some event_type",
+        start_date: ~N[2023-10-05 06:18:00],
+        end_date: ~N[2023-10-05 06:18:00]
+      }
+
+      assert {:ok, %Event{} = event} = Events.create_event(valid_attrs)
```
Comment (okothkongo) [inline test/elixir_conf_africa/events_test.exs:56]: There is alot of redundancy here maybe we should introduce factories.

WDYT?

**Review by okothkongo — CHANGES_REQUESTED** [review]

File: test/elixir_conf_africa/events_test.exs (hunk 2 above)
Comment (MICHAELMUNAVU83) [reply test/elixir_conf_africa/events_test.exs:56]: I have not interacted with them before , is it this https://hexdocs.pm/ecto/test-factories.html ?

Comment (okothkongo) [reply test/elixir_conf_africa/events_test.exs:56]: Precisely

File: lib/elixir_conf_africa/events.ex (hunk 1 above)
Comment (MICHAELMUNAVU83) [reply lib/elixir_conf_africa/events.ex]: Fixed this

File: test/elixir_conf_africa/events_test.exs (hunk 2 above)
Comment (MICHAELMUNAVU83) [reply test/elixir_conf_africa/events_test.exs]: Hey @okothkongo   , Thanks for this , I read it over the weekend and got to understand its benefits , I have implimented it in the changes below , Thanks

**Review by okothkongo — CHANGES_REQUESTED** [review]
What makes us sure that inserted event is the same as the one we get with `get_elixir_conf_event /0`.

What if we insert two events with the same name 🤔

File: test/elixir_conf_africa/events_test.exs [hunk 3]
```diff
@@ -22,11 +24,38 @@ defmodule ElixirConfAfrica.EventsTest do
...
+    end
+
+    test "get_all_available_tickets/0 returns the number of available tickets" do
+      event =
+        insert!(:elixir_conf_event)
+
+      ticket_type =
+        insert!(:elixir_conf_ticket_type, event_id: event.id)
+
+      assert Events.get_all_available_tickets() == ticket_type.number
```
Comment (okothkongo) [inline test/elixir_conf_africa/events_test.exs:56]: Maybe rename this function to `get_number_of_tickets_available` with the current it suggests maybe we are getting a list of tickets.

also believe we should give this function parameter to make this more dynamic.

```  def get_all_available_tickets do
    query =
      from t in TicketType,
        join: e in Event,
        on: t.event_id == e.id and e.name == "ElixirConf Africa 2024",
        select: sum(t.number)

    Repo.one(query)
  end
  ```
  But if we want a specific event name say in this case `ElixirConf Africa 2024` then
  the name should be `get_number_of_tickets_available_for_elixir_conf_2024`(long 😄  but explicit)

**Comment by MICHAELMUNAVU83** [comment]
> What makes us sure that the inserted event is of the same `name` as the one we get with `get_elixir_conf_event /0`?
> 
> What if we insert two events with the same name 🤔

@okothkongo  , we can add a unique constraint to the name , would this suffice?

File: test/elixir_conf_africa/events_test.exs (hunk 3 above)
Comment (MICHAELMUNAVU83) [reply test/elixir_conf_africa/events_test.exs:56]: I agree , I have changed the function name to  `get_number_of_tickets_available_for_elixir_conf_2024` and added documentation for the same

**Pull request merged**